include/util/guidance/name_announcements.hpp

```diff
@@ -160,7 +160,9 @@ inline bool requiresNameAnnounced(const StringView &from_name,
     const auto needs_announce =
         // " (Ref)" -> "Name " and reverse
         (from_name.empty() && !from_ref.empty() && !to_name.empty() && to_ref.empty()) ||
-        (!from_name.empty() && from_ref.empty() && to_name.empty() && !to_ref.empty());
+        (!from_name.empty() && from_ref.empty() && to_name.empty() && !to_ref.empty()) ||
+        // ... or names are empty but reference changed
+        (names_are_empty && !ref_is_contained);
```

Comment:
👍 I was concerned that the `ref_is_contained` check would be insufficient in cases where a route follows an exit and an auxiliary route continues straight. However, the interchanges at either end of [this route](https://www.openstreetmap.org/relation/1933421) (`I 40` versus `I 40 Business`) demonstrate that the suffixed cardinal directions prevent `ref_is_contained` from being true. (In this case, the main route exits as a `motorway`, so there’s a fork instruction anyways, but I’m sure there are cases where the main route exits as a `motorway_link`.)